openedx_learning/core/publishing/model_mixins.py

```diff
@@ -129,59 +134,110 @@ def __init__(self, content_obj):
             # get the reverse related field name so that we can use that later.
             self.related_name = field_to_pev.related_query_name()
 
+        def _content_obj_version(self, pub_ent_version: PublishableEntityVersion | None):
+            """
+            PublishableEntityVersion -> Content object version
+
+            Given a reference to a PublishableEntityVersion, return the version
+            of the content object that we've been mixed into.
+            """
+            if pub_ent_version is None:
+                return None
+            return getattr(pub_ent_version, self.related_name)
+
         @property
         def draft(self):
             """
             Return the content version object that is the current draft.
+
+            So if you mix ``PublishableEntityMixin`` into ``Component``, then
+            ``component.versioning.draft`` will return you the
+            ``ComponentVersion`` that is the current draft (not the underlying
+            ``PublishableEntityVersion``).
+
+            If this is causing many queries, it might be the case that you need
+            to add ``select_related('publishable_entity__draft__version')`` to
+            the queryset.
```

Comment:
If you want, [you can use `Field.is_cached` to check this in code with zero db overhead](https://stackoverflow.com/a/68200269), and display a warning to the developer at runtime. But though it's not a private method, it's technically undocumented functionality.

Reply:
Thanks. I think I'll avoid it for now, but I'll keep it in mind if performance regressions along these lines starts to become a problem.